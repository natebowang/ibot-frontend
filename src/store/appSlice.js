import {createSlice} from "@reduxjs/toolkit";

/**
 *  States for the whole application.
 */
const appSlice = createSlice({
    name: 'app',
    initialState: {
        wiw: window.innerWidth,
    },
    reducers: {
        updateWiw: state => {
            state.wiw = window.innerWidth
        },
    },
});

export const {updateWiw} = appSlice.actions;

export const selectWiw = state => state.app.wiw;

export const appReducer = appSlice.reducer;
