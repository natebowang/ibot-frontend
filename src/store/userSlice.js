import {createSlice} from "@reduxjs/toolkit";

const userSlice = createSlice({
    name: 'user',
    initialState: {
        username: '',
        avatar: null,
        incorrectUsernamePwdFlag: false,
    },
    reducers: {
        updateUserProfile: (state, action) => {
            state.username = action.payload.username;
            state.avatar = action.payload.avatar;
        },
        updateIncorrectUsernamePwdFlag: (state, action) => {
            state.incorrectUsernamePwdFlag = action.payload.incorrectUsernamePwdFlag;
        }
    }
});

export const {updateUserProfile, updateIncorrectUsernamePwdFlag} = userSlice.actions;
export const userReducer = userSlice.reducer;

export const login = (userProfile) => async (dispatch) => {
    // const response = await fetch('./user.json', {
    //     method: 'POST',
    //     headers: {
    //         "Content-Type": "application/json",
    //     },
    //     body: JSON.stringify(userProfile),
    // });
    const response = await fetch('./user.json');
    const message = await response.json();
    if (response.ok) {
        console.log(message);
    } else {
        console.log('error');
    }
};

export const signup = (userProfile) => async (dispatch) => {
    const response = await fetch('./api/signup', {
        method: 'POST',
        headers: {
            "Content-Type": "application/json",
        },
        body: JSON.stringify(userProfile),
    });
    const message = await response.json();
    if (response.ok) {
        console.log(message);
    } else {
        console.log('error');
    }
};
export const selectUsername = state => state.user.username;
export const selectAvatar = state => state.user.avatar;
export const selectIncorrectUsernamePwdFlag = state => state.user.incorrectUsernamePwdFlag;

