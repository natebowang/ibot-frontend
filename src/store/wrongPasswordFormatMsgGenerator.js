const wrongPasswordFormatMsgGenerator = (passwordValidationResultArray) => {
    switch (true) {
        case passwordValidationResultArray.indexOf('min') > -1:
            // return 'Your password needs 6+ characters. ';
            return '需包含6个以上字符。';
        case passwordValidationResultArray.indexOf('max') > -1:
            // return 'Your password must have no more than 50 characters. ';
            return '需少于50个字符。';
        case passwordValidationResultArray.indexOf('digits') > -1:
            // return 'Your password must have 1+ digit. ';
            return '必须包含至少一个数字。';
        case passwordValidationResultArray.indexOf('symbols') > -1:
            // return 'Your password must have 1+ symbol. ';
            return '必须包含至少一个符号。';
        case passwordValidationResultArray.indexOf('space') > -1:
            // return 'Your password must not have space. ';
            return '不能包含空格。';
        default:
            // return 'Wrong password format. ';
            return '密码格式错误。';
    }
};

export default wrongPasswordFormatMsgGenerator;
