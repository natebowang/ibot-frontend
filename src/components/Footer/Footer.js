import React from 'react';
import qr from './mokeQR.jpg';
import {useSelector} from "react-redux";
import {selectWiw} from "../../store/appSlice";
import {FOOTER_HEIGHT, MOBILE_PORTRAIT_WIDTH} from "../../app/config";
import {NavLink} from "react-router-dom";

const Footer = () => {
    const wiw = useSelector(selectWiw);
    return (
        <footer style={footerStyle}>
            <div style={boxStyle}>
                {wiw > MOBILE_PORTRAIT_WIDTH ? <Links/> : null}
                <div style={itemStyle}>
                    <h2 style={boldLinksStyle}>微信关注</h2>
                    <p style={normalLinksStyle}>
                        微信 ：caomin19690307
                    </p>
                    <img src={qr} alt={'QR'} style={qrStyle}/>
                </div>
                <div style={itemStyle}>
                    <h2 style={boldLinksStyle}>联系我们</h2>
                    <p style={normalLinksStyle}>
                        电话：0279689953
                    </p>
                    <p style={normalLinksStyle}>
                        邮箱：cx4648@gmail.com
                    </p>
                </div>
            </div>
        </footer>
    )
};

const Links = () => (
    <>
        <div style={itemStyle}>
            <NavLink to={'./teachers'} style={boldLinksStyle}>教师</NavLink>
            <NavLink to={'./teachers'} style={normalLinksStyle}>陈教师</NavLink>
            <NavLink to={'./teachers'} style={normalLinksStyle}>王教师</NavLink>
        </div>
        <div style={itemStyle}>
            <NavLink to={'./classes'} style={boldLinksStyle}>课程</NavLink>
            <NavLink to={'./classes'} style={normalLinksStyle}>Lego搭建</NavLink>
            <NavLink to={'./classes'} style={normalLinksStyle}>机器人</NavLink>
        </div>
        <div style={itemStyle}>
            <NavLink to={'./championship'} style={boldLinksStyle}>比赛</NavLink>
            <NavLink to={'./championship'} style={normalLinksStyle}>Robocup新西兰赛</NavLink>
            <NavLink to={'./championship'} style={normalLinksStyle}>Robocup澳洲赛</NavLink>
            <NavLink to={'./championship'} style={normalLinksStyle}>Robocup国际赛</NavLink>
        </div>
    </>
);

const footerStyle = {
    backgroundColor: 'black',
    color: 'white',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: '1rem',
    height: FOOTER_HEIGHT,
};

const boxStyle = {
    width: '100%',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'flex-start',
};

const itemStyle = {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
};

const boldLinksStyle = {
    color: 'white',
    textDecoration: 'none',
    textAlign: 'center',
    padding: '0',
    margin: '0.8rem 1rem',
    fontSize: '0.8rem',
    fontWeight: 'bold',
};

const normalLinksStyle = {
    color: 'white',
    textDecoration: 'none',
    textAlign: 'left',
    padding: '0',
    margin: '0rem 0rem',
    lineHeight: '1.8rem',
    fontSize: '0.7rem',
    fontWeight: 'bold',
};

const qrStyle = {
    width: '60px',
    height: 'auto',
    margin: '1.2rem 0rem',
};
export default Footer;