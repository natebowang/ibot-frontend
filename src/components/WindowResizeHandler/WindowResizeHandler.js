import {useEffect} from 'react';
import {useDispatch} from "react-redux";
import {updateWiw} from "../../store/appSlice";

/**
 * Window resize event handler component. It updates store when window resized.
 */
const WindowResizeHandler = () => {
    const dispatch = useDispatch();

    const windowResizeHandler = (e) => {
        dispatch(updateWiw());
    };

    useEffect(() => {
        window.addEventListener("resize", windowResizeHandler);
        return () => {
            window.removeEventListener("resize", windowResizeHandler)
        };
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    return null;
};

export default WindowResizeHandler;