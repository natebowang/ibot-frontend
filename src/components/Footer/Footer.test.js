import React from "react";
import TestRenderer from 'react-test-renderer';
import Footer from "./Footer";
import {Provider} from "react-redux";
import store from "../../store/store";
import {BrowserRouter} from "react-router-dom";
import {StyleRoot} from "radium";
import WindowResizeHandler from "../WindowResizeHandler/WindowResizeHandler";

const {act} = TestRenderer;

const footer = TestRenderer.create(
    <Provider store={store}>
        <BrowserRouter>
            <StyleRoot>
                <WindowResizeHandler/>
                <Footer/>
            </StyleRoot>
        </BrowserRouter>
    </Provider>
);

test('Desktop footer component snapshot test', () => {
    act(() => {
        window.innerWidth = 1024;
        window.dispatchEvent(new Event('resize'));
    });
    const tree = footer.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Mobile footer component snapshot test', () => {
    act(() => {
        window.innerWidth = 300;
        window.dispatchEvent(new Event('resize'));
    });
    const tree = footer.toJSON();
    expect(tree).toMatchSnapshot();
});
