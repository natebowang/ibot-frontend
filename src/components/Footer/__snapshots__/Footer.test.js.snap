// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop footer component snapshot test 1`] = `
<div
  data-radium={true}
>
  <footer
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "color": "white",
        "display": "flex",
        "flexDirection": "column",
        "height": "12.5rem",
        "padding": "1rem",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <a
          aria-current={null}
          href="/teachers"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "fontSize": "0.8rem",
              "fontWeight": "bold",
              "margin": "0.8rem 1rem",
              "padding": "0",
              "textAlign": "center",
              "textDecoration": "none",
            }
          }
        >
          教师
        </a>
        <a
          aria-current={null}
          href="/teachers"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "fontSize": "0.7rem",
              "fontWeight": "bold",
              "lineHeight": "1.8rem",
              "margin": "0rem 0rem",
              "padding": "0",
              "textAlign": "left",
              "textDecoration": "none",
            }
          }
        >
          陈教师
        </a>
        <a
          aria-current={null}
          href="/teachers"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "fontSize": "0.7rem",
              "fontWeight": "bold",
              "lineHeight": "1.8rem",
              "margin": "0rem 0rem",
              "padding": "0",
              "textAlign": "left",
              "textDecoration": "none",
            }
          }
        >
          王教师
        </a>
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <a
          aria-current={null}
          href="/classes"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "fontSize": "0.8rem",
              "fontWeight": "bold",
              "margin": "0.8rem 1rem",
              "padding": "0",
              "textAlign": "center",
              "textDecoration": "none",
            }
          }
        >
          课程
        </a>
        <a
          aria-current={null}
          href="/classes"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "fontSize": "0.7rem",
              "fontWeight": "bold",
              "lineHeight": "1.8rem",
              "margin": "0rem 0rem",
              "padding": "0",
              "textAlign": "left",
              "textDecoration": "none",
            }
          }
        >
          Lego搭建
        </a>
        <a
          aria-current={null}
          href="/classes"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "fontSize": "0.7rem",
              "fontWeight": "bold",
              "lineHeight": "1.8rem",
              "margin": "0rem 0rem",
              "padding": "0",
              "textAlign": "left",
              "textDecoration": "none",
            }
          }
        >
          机器人
        </a>
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <a
          aria-current={null}
          href="/championship"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "fontSize": "0.8rem",
              "fontWeight": "bold",
              "margin": "0.8rem 1rem",
              "padding": "0",
              "textAlign": "center",
              "textDecoration": "none",
            }
          }
        >
          比赛
        </a>
        <a
          aria-current={null}
          href="/championship"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "fontSize": "0.7rem",
              "fontWeight": "bold",
              "lineHeight": "1.8rem",
              "margin": "0rem 0rem",
              "padding": "0",
              "textAlign": "left",
              "textDecoration": "none",
            }
          }
        >
          Robocup新西兰赛
        </a>
        <a
          aria-current={null}
          href="/championship"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "fontSize": "0.7rem",
              "fontWeight": "bold",
              "lineHeight": "1.8rem",
              "margin": "0rem 0rem",
              "padding": "0",
              "textAlign": "left",
              "textDecoration": "none",
            }
          }
        >
          Robocup澳洲赛
        </a>
        <a
          aria-current={null}
          href="/championship"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
              "fontSize": "0.7rem",
              "fontWeight": "bold",
              "lineHeight": "1.8rem",
              "margin": "0rem 0rem",
              "padding": "0",
              "textAlign": "left",
              "textDecoration": "none",
            }
          }
        >
          Robocup国际赛
        </a>
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <h2
          style={
            Object {
              "color": "white",
              "fontSize": "0.8rem",
              "fontWeight": "bold",
              "margin": "0.8rem 1rem",
              "padding": "0",
              "textAlign": "center",
              "textDecoration": "none",
            }
          }
        >
          微信关注
        </h2>
        <p
          style={
            Object {
              "color": "white",
              "fontSize": "0.7rem",
              "fontWeight": "bold",
              "lineHeight": "1.8rem",
              "margin": "0rem 0rem",
              "padding": "0",
              "textAlign": "left",
              "textDecoration": "none",
            }
          }
        >
          微信 ：caomin19690307
        </p>
        <img
          alt="QR"
          src="mokeQR.jpg"
          style={
            Object {
              "height": "auto",
              "margin": "1.2rem 0rem",
              "width": "60px",
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <h2
          style={
            Object {
              "color": "white",
              "fontSize": "0.8rem",
              "fontWeight": "bold",
              "margin": "0.8rem 1rem",
              "padding": "0",
              "textAlign": "center",
              "textDecoration": "none",
            }
          }
        >
          联系我们
        </h2>
        <p
          style={
            Object {
              "color": "white",
              "fontSize": "0.7rem",
              "fontWeight": "bold",
              "lineHeight": "1.8rem",
              "margin": "0rem 0rem",
              "padding": "0",
              "textAlign": "left",
              "textDecoration": "none",
            }
          }
        >
          电话：0279689953
        </p>
        <p
          style={
            Object {
              "color": "white",
              "fontSize": "0.7rem",
              "fontWeight": "bold",
              "lineHeight": "1.8rem",
              "margin": "0rem 0rem",
              "padding": "0",
              "textAlign": "left",
              "textDecoration": "none",
            }
          }
        >
          邮箱：cx4648@gmail.com
        </p>
      </div>
    </div>
  </footer>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`Mobile footer component snapshot test 1`] = `
<div
  data-radium={true}
>
  <footer
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "color": "white",
        "display": "flex",
        "flexDirection": "column",
        "height": "12.5rem",
        "padding": "1rem",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <h2
          style={
            Object {
              "color": "white",
              "fontSize": "0.8rem",
              "fontWeight": "bold",
              "margin": "0.8rem 1rem",
              "padding": "0",
              "textAlign": "center",
              "textDecoration": "none",
            }
          }
        >
          微信关注
        </h2>
        <p
          style={
            Object {
              "color": "white",
              "fontSize": "0.7rem",
              "fontWeight": "bold",
              "lineHeight": "1.8rem",
              "margin": "0rem 0rem",
              "padding": "0",
              "textAlign": "left",
              "textDecoration": "none",
            }
          }
        >
          微信 ：caomin19690307
        </p>
        <img
          alt="QR"
          src="mokeQR.jpg"
          style={
            Object {
              "height": "auto",
              "margin": "1.2rem 0rem",
              "width": "60px",
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <h2
          style={
            Object {
              "color": "white",
              "fontSize": "0.8rem",
              "fontWeight": "bold",
              "margin": "0.8rem 1rem",
              "padding": "0",
              "textAlign": "center",
              "textDecoration": "none",
            }
          }
        >
          联系我们
        </h2>
        <p
          style={
            Object {
              "color": "white",
              "fontSize": "0.7rem",
              "fontWeight": "bold",
              "lineHeight": "1.8rem",
              "margin": "0rem 0rem",
              "padding": "0",
              "textAlign": "left",
              "textDecoration": "none",
            }
          }
        >
          电话：0279689953
        </p>
        <p
          style={
            Object {
              "color": "white",
              "fontSize": "0.7rem",
              "fontWeight": "bold",
              "lineHeight": "1.8rem",
              "margin": "0rem 0rem",
              "padding": "0",
              "textAlign": "left",
              "textDecoration": "none",
            }
          }
        >
          邮箱：cx4648@gmail.com
        </p>
      </div>
    </div>
  </footer>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;
